.travis.yml

```diff
@@ -2,15 +2,21 @@ language: cpp
 compiler:
     - gcc
 env:
-    matrix:
-        - FIREFOXVERSION="36.0.1"
-        - FIREFOXVERSION="31.5.0esr"
+    - FX_PATH_SUFFIX=""
+    - FX_PATH_SUFFIX="-esr"
+    - FX_PATH_SUFFIX="-beta"
+#    - FX_PATH_SUFFIX="-prior-esr"
+matrix:
+    fast_finish: true
+    allow_failures:
+        - env: FX_PATH_SUFFIX="-beta"
+        - env: FX_PATH_SUFFIX="-prior-esr"
 notifications:
     email: false
 before_install:
     - export DISPLAY=:99.0
     - sh -e /etc/init.d/xvfb start
-    - wget http://ftp.mozilla.org/pub/firefox/releases/${FIREFOXVERSION}/linux-x86_64/en-US/firefox-${FIREFOXVERSION}.tar.bz2
-    - tar -xjf firefox-${FIREFOXVERSION}.tar.bz2
+    - wget ftp://ftp.mozilla.org/pub/mozilla.org/firefox/releases/latest${FX_PATH_SUFFIX}/linux-x86_64/en-US/firefox-*.tar.bz2
```

Comment:
This was HTTP on purpose, so that Travis could potentially cache it. (FTP caching seems less likely.) We could make a separate GET request to the directory index to get the version number, store that in a variable, and then request that via HTTP without the wildcard.